script/run-comp-match

```diff
@@ -0,0 +1,200 @@
+
+
+def run_match() -> None:
+    try:
+        subprocess.check_call([
+            'webots',
+            '--batch',
+            '--stdout',
+            '--stderr',
+            '--minimize',
+            str(REPO_ROOT / 'worlds' / 'Arena.wbt'),
+        ])
+    except subprocess.CalledProcessError as e:
+        BOLD = '\033[1m'
+        FAIL = '\033[91m'
+        ENDC = '\033[0m'
+
+        log_text = controller_utils.get_competition_supervisor_log_filepath().read_text()
+
+        print("\n" + FAIL + log_text + ENDC)  # noqa:T001
+        print(  # noqa:T001
+            f"{BOLD}{FAIL}Simulation errored (exit code {e.returncode}). "
+            f"Competition supervisor logs are above.{ENDC}",
+        )
+        exit(1)
+
+
+def collate_logs(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    """
+    Copy the teams' logs into directories next to their original code archives.
+    """
+
+    for zone_id, tla in enumerate(match_data.teams):
+        if tla is None:
+            # no team in this zone
+            continue
+
+        log_filname = controller_utils.get_robot_log_filename(zone_id)
+        log_path = get_zone_path(zone_id) / log_filname
+
+        team_dir = archives_dir / tla
+        team_dir.mkdir(exist_ok=True)
+
+        shutil.copy(log_path, team_dir)
+
+
+def archive_match_file(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    """
+    Copy the match file (which contains the scoring data) to the archives directory.
+
+    This also renames the file to the name which SRComp expects.
+    """
+
+    matches_dir = archives_dir / 'matches'
+    matches_dir.mkdir(exist_ok=True)
+
+    # The file contains JSON data, we're relying here on the fact that JSON is a
+    # strict subset of YAML.
+    completed_match_file = matches_dir / f'{match_data.match_number:0>3}.yaml'
+
+    shutil.copy(controller_utils.get_match_file(), completed_match_file)
+
+
+def archive_match_recordings(archives_dir: Path) -> None:
+    """
+    Copy the recordings to the archives directory.
+    """
+
+    recordings_dir = archives_dir / 'recordings'
+    recordings_dir.mkdir(exist_ok=True)
+
+    recording_stem = controller_utils.get_recording_stem()
+
+    for path in recording_stem.parent.glob(f'{recording_stem.name}.*'):
+        shutil.copy(path, recordings_dir)
+
+    try:
+        shutil.copytree(recording_stem.parent / 'textures', recordings_dir / 'textures')
+    except FileExistsError:
+        pass
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory containing the teams' robot code, as Zip archives "
+            "named for the teams' TLAs. This directory will also be used as the "
+            "root for storing the resulting logs and recordings."
+        ),
+        type=Path,
+    )
+    parser.add_argument('match_num', help="The number of the match to run.")
```

Comment:
We don't appear to be validating that the match number is an int anywhere.
```suggestion
    parser.add_argument(
        'match_num',
        help="The number of the match to run.",
        type=int,
    )
```

Reply:
Good spot :) 0259858